pages/_app.js

```diff
@@ -1,28 +1,37 @@
 import GlobalStyle from "../styles";
 import Image from "next/image";
-import styled from "styled-components";
+import plants from "/lib/data";
+import useLocalStorageState from "use-local-storage-state";
 
 export default function App({ Component, pageProps }) {
-  
+  const [allPlants, setAllPlants] = useLocalStorageState("allPlants", {
+    defaultValue: plants,
+  });
+
+  function handleToggleOwned(id) {
+    setAllPlants((prevPlants) =>
+      prevPlants.map((plant) =>
+        plant.id === id ? { ...plant, isOwned: !plant.isOwned } : plant
+      )
+    );
+  }
+
   return (
     <>
       <GlobalStyle />
-      <StyledHeader>
-        <Image 
+      <header>
+        <Image
           src={"/logo-main.svg"}
```

Comment:
_Just a tiny note:_ The curly braces `{}` are not needed for plain strings in JSX attributes.

```suggestion
          src="/logo-main.svg"
```